drv/ksz8463/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+#![no_std]
+
+use core::convert::TryInto;
+
+use drv_spi_api::{SpiDevice, SpiError};
+use drv_stm32h7_gpio_api as gpio_api;
+use gpio_api::{Gpio, OutputType, PinSet, Pull, Speed};
+use ringbuf::*;
+use userlib::{hl::sleep_for, task_slot};
+
+task_slot!(GPIO, gpio_driver);
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+enum Trace {
+    None,
+    Read(Register, u16),
+    Write(Register, u16),
+    Id(u16),
+}
+ringbuf!(Trace, 16, Trace::None);
+
+const fn register_offset(address: u16) -> u16 {
+    let addr10_2 = address >> 2;
+    let mask_shift = 2 /* turn around bits */ + (2 * ((address >> 1) & 0x1));
+    (addr10_2 << 6) | ((0x3 as u16) << mask_shift)
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+#[repr(u16)]
```

Comment:
Nit: do you need this? We've got a lot of enums with explicit `repr`s but it's usually to support deriving `zerocopy`, which you're not.

Reply:
It appears to be needed, otherwise I get errors of the form
```
error[E0308]: mismatched types
  --> drv/ksz8463/src/lib.rs:32:13
   |
32 |     CIDER = register_offset(0x000),
   |             ^^^^^^^^^^^^^^^^^^^^^^ expected `isize`, found `u16`
```

I want to do a second pass through this, either before or after merging, since this is left over from Arjen's original `100base-fx` branch and doesn't match in style.

Reply:
I ended up reworking where this conversion happens, so the `repr(u16)` is no longer needed